plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerMachineSource.java

```diff
@@ -0,0 +1,226 @@
+     * @return current instance
+     */
+    public void setTag(String tag) {
+        this.tag = tag;
+    }
+
+    /**
+     * Defines optional registry attribute
+     * @param registry as for example docker-registry.company.com:5000
+     */
+    public void setRegistry(String registry) {
+        this.registry = registry;
+    }
+
+    /**
+     * Defines optional registry attribute
+     * @param registry as for example docker-registry.company.com:5000
+     * @return current instance
+     */
+    public DockerMachineSource withRegistry(String registry) {
+        this.registry = registry;
+        return this;
+    }
+
+    /**
+     * Defines optional digest attribute
+     * @param digest as for example sha256@1234
+     */
+    public void setDigest(String digest) {
+        this.digest = digest;
+    }
+
+    /**
+     * Defines optional digest attribute
+     * @param digest as for example sha256@1234
+     * @return current instance
+     */
+    public DockerMachineSource withDigest(String digest) {
+        this.digest = digest;
+        return this;
+    }
+
+    /**
+     * @return mandatory repository
+     */
+    public String getRepository() {
+        return this.repository;
+    }
+
+    /**
+     * @return optional tag
+     */
+    public String getTag() {
+        return this.tag;
+    }
+
+    /**
+     * @return optional registry
+     */
+    public String getRegistry() {
+        return this.registry;
+    }
+
+    /**
+     * @return optional digest
+     */
+    public String getDigest() {
+        return this.digest;
+    }
+
+    /**
+     * Returns location of this docker image, including all data that are required to reconstruct a new docker machine source.
+     */
+    public String getLocation() {
+        return getLocation(true);
+    }
+
+    /**
+     * Returns full name of docker image.
+     * <p>
+     * It consists of registry, userspace, repository name, tag, digest.
+     * E.g. docker-registry.company.com:5000/userspace1/my-repository:some-tag
+     * E.g. docker-registry.company.com:5000/userspace1/my-repository@some-digest
+     * @param includeDigest if digest needs to be included or not
+     */
+    public String getLocation(boolean includeDigest) {
+        final StringBuilder fullRepoId = new StringBuilder();
+
+        // optional registry is followed by /
+        if (getRegistry() != null) {
+            fullRepoId.append(getRegistry()).append('/');
+        }
+
+        // repository
+        fullRepoId.append(getRepository());
+
+        // optional tag (: prefix)
+        if (getTag() != null) {
+            fullRepoId.append(':').append(getTag());
+        }
+
+        // optional digest (@ prefix)
+        if (includeDigest && getDigest() != null) {
+            fullRepoId.append('@').append(getDigest());
+        }
+        return fullRepoId.toString();
+    }
+
+    /**
+     * Returns full name of docker image (without digest)
+     */
+    public String getFullName() {
```

Comment:
But you changed the name of the class so no backward compliance is supposed I think.